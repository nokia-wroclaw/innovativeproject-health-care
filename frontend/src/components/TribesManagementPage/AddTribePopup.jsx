import React from "react";
import { Button, Icon, Popup } from "semantic-ui-react";
import "../../styles/common.css";
import AddTribeForm from "./AddTribeForm";

export const AddTribePopup = () => {
  return (
    <Popup
      trigger={
        <Button icon labelPosition="left" floated="right" compact secondary>
          <Icon name="plus" />
          New tribe
        </Button>
      }
      content={<AddTribeForm />}
      on="click"
      position="bottom center"
    />
  );
};

export default AddTribePopup;
