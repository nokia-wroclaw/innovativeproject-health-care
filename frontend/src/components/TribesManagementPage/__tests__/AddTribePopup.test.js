import React from "react";
import { shallow } from "enzyme";
import { AddTribePopup } from "../AddTribePopup";

describe("AddTribePopup", () => {
  const wrapper = shallow(<AddTribePopup />);

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
