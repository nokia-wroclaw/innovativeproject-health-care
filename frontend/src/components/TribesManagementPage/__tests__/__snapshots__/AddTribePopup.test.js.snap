// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTribePopup should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddTribePopup />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "content": <Connect(AddTribeForm) />,
      "disabled": false,
      "keepInViewPort": true,
      "on": "click",
      "position": "bottom center",
      "trigger": <Button
        as="button"
        compact={true}
        floated="right"
        icon={true}
        labelPosition="left"
        secondary={true}
      >
        <Icon
          as="i"
          name="plus"
        />
        New tribe
      </Button>,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "content": <Connect(AddTribeForm) />,
        "disabled": false,
        "keepInViewPort": true,
        "on": "click",
        "position": "bottom center",
        "trigger": <Button
          as="button"
          compact={true}
          floated="right"
          icon={true}
          labelPosition="left"
          secondary={true}
        >
          <Icon
            as="i"
            name="plus"
          />
          New tribe
        </Button>,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
